src/models/tabularmodel.jl

```diff
@@ -0,0 +1,79 @@
+function emb_sz_rule(n_cat)
+	min(600, round(1.6 * n_cat^0.56))
+end
+
+function _one_emb_sz(catdict, catcol::Symbol, sz_dict=nothing)
+	sz_dict = isnothing(sz_dict) ? Dict() : sz_dict
+	n_cat = length(catdict[catcol])
+	sz = catcol in keys(sz_dict) ? sz_dict[catcol] : emb_sz_rule(n_cat)
+	n_cat, sz
+end
+
+function get_emb_sz(catdict, cols, sz_dict=nothing)
+	[_one_emb_sz(catdict, catcol, sz_dict) for catcol in cols]
+end
+
+# function get_emb_sz(td::TableDataset, sz_dict=nothing)
+# 	cols = Tables.columnaccess(td.table) ? Tables.columnnames(td.table) : Tables.columnnames(Tables.rows(td.table)[1])
+# 	[_one_emb_sz(catdict, catcol, sz_dict) for catcol in cols]
+# end
+
+struct TabularModel
+	embeds
+	emb_drop
+	bn_cont
+	n_emb
+	n_cont
+	layers
+end
+
+function TabularModel(
+		layers; 
+		emb_szs,
+		n_cont,
+		out_sz,
+		ps::Union{Tuple, Vector, Number, Nothing}=nothing,
+		embed_p::Float64=0.,
+		y_range=nothing,
+		use_bn::Bool=true,
+		bn_final::Bool=false,
+		bn_cont::Bool=true,
+		act_cls=Flux.relu,
+		lin_first::Bool=true)
+
+	n_cont = Int64(n_cont)
+	if isnothing(ps)
+		ps = zeros(length(layers))
+	end
+	if ps isa Number
+		ps = fill(ps, length(layers))
+	end
+	embedslist = [Embedding(ni, nf) for (ni, nf) in emb_szs]
+	emb_drop = Dropout(embed_p)
+	bn_cont = bn_cont ? BatchNorm(n_cont) : false
+	n_emb = sum(size(embedlayer.weight)[1] for embedlayer in embedslist)
+	sizes = append!(zeros(0), [n_emb+n_cont], layers, [out_sz])
+	actns = append!([], [act_cls for i in 1:(length(sizes)-1)], [nothing])
+	_layers = [linbndrop(Int64(sizes[i]), Int64(sizes[i+1]), use_bn=(use_bn && ((i!=(length(actns)-1)) || bn_final)), p=p, act=a, lin_first=lin_first) for (i, (p, a)) in enumerate(zip(push!(ps, 0.), actns))]
+	if !isnothing(y_range)
+		push!(_layers, Chain(@. x->Flux.sigmoid(x) * (y_range[2] - y_range[1]) + y_range[1]))
+	end
+	layers = Chain(_layers...)
+	TabularModel(embedslist, emb_drop, bn_cont, n_emb, n_cont, layers)
+end
+
+function (tm::TabularModel)(x)
+	x_cat, x_cont = x
+	if tm.n_emb != 0
+		x = [e(x_cat[i, :]) for (i, e) in enumerate(tm.embeds)]
+		x = vcat(x...)
+		x = tm.emb_drop(x)
+	end
+	if tm.n_cont != 0
+		if (tm.bn_cont != false)
```

Comment:
Equivalent to `if tm.bn_cont == true` which is equivalent to `if tm.bn_cont`.
```suggestion
		if tm.bn_cont
```